nearcore/src/config.rs

```diff
@@ -1098,32 +1099,71 @@ pub fn get_config_url(chain_id: &String) -> String {
-    actix::System::new().block_on(async move {
-        let client = awc::Client::new();
-        let mut response = client.get(url).send().await.expect("Unable to download the file");
-        // IMPORTANT: limit specifies the maximum size of the genesis or config file
-        // In case where the genesis or config file is bigger than the specified
-        // limit Overflow Error is thrown
-        let body = response
-            .body()
-            .limit(limit)
-            .await
-            .expect("File is bigger than specified limit. Please make the limit higher.");
-
-        std::fs::write(&path, &body).expect("Failed to create / write a file.");
+#[derive(thiserror::Error, Debug)]
+pub enum FileDownloadError {
+    #[error("Failed to download the file: {0}")]
+    HttpError(#[from] hyper::Error),
+    #[error("Failed to open file: {0}")]
+    OpenError(std::io::Error),
+    #[error("Failed to write to file: {0}")]
+    WriteError(std::io::Error),
+    #[error("Failed to rename file: {0}")]
+    RenameError(std::io::Error),
+    #[error("Invalid URI: {0}")]
+    UriError(#[from] hyper::http::uri::InvalidUri),
+    #[error("Failed to remove the temporary file after failure: {0}, {1}")]
+    RemoveTemporaryFileError(std::io::Error, Box<FileDownloadError>)
+}
+
+pub fn download_file(url: &String, path: &Path) -> Result<(), FileDownloadError> {
+    return actix::System::new().block_on(async move {
+        let https_connector = hyper_tls::HttpsConnector::new();
+        let client = hyper::Client::builder().build::<_, hyper::Body>(https_connector);
+
+        let uri = url.parse()?;
+        let mut resp = client.get(uri).await?;
+
+        // To avoid partially downloaded files, we first download the file to a temporary *.swp file
+        // and rename it once the download is finished.
+        let tmp_path = path.with_extension("swp");
+        let mut tmp_file = File::create(&tmp_path).map_err(|e| {
+            FileDownloadError::OpenError(e)
+        })?;
+
+        let process_tmp_file = async {
+            while let Some(next_chunk_result) = resp.data().await {
+                let next_chunk = next_chunk_result?;
+                tmp_file.write_all(next_chunk.as_ref()).map_err(|e| {
+                    FileDownloadError::WriteError(e)
+                })?;
+            }
+            std::fs::rename(&tmp_path, path).map_err(|e| {
+                FileDownloadError::RenameError(e)
+            })?;
+            Ok(())
+        };
+
+        // We run everything that interacts with the temporary *.swp file in a separate async block
+        // so that we can clean up and safely delete it if something went wrong.
+        process_tmp_file.await.map_err(|e| match std::fs::remove_file(tmp_path) {
+            Ok(_) => e,
+            Err(remove_file_error) => {
+                FileDownloadError::RemoveTemporaryFileError(remove_file_error, Box::new(e))
+            }
+        })?;
+        Ok(())
```

Comment:
nit: already returning `Ok(())` in the `process_tmp_file` block, so we can just forward it at the end instead
```suggestion
        process_tmp_file.await.map_err(|e| match std::fs::remove_file(tmp_path) {
            Ok(_) => e,
            Err(remove_file_error) => {
                FileDownloadError::RemoveTemporaryFileError(remove_file_error, Box::new(e))
            }
        })
```